File: android/PlayKit/app/src/main/java/io/streamroot/dna/playkit/MainActivity.java
package io.streamroot.dna.playkit;

import android.content.Intent;
import android.os.Bundle;
import android.widget.AutoCompleteTextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.launchButton).setOnClickListener(v -> {
            Intent intent = new Intent(getApplicationContext(), PlayerActivity.class);
            AutoCompleteTextView streamEditText = findViewById(R.id.streamEditText);

            intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            intent.putExtra("streamUrl", streamEditText.getText().toString());

            getApplicationContext().startActivity(intent);
        });
    }
}

File: android/ExoPlayer/app/src/main/java/io/streamroot/dna/exoplayer/MainActivity.kt
package io.streamroot.dna.exoplayer

import android.content.Intent
import android.os.Bundle
import android.widget.AutoCompleteTextView
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var streamEditText: AutoCompleteTextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        streamEditText = findViewById(R.id.streamEditText)

        findViewById<Button>(R.id.launchButton).setOnClickListener {
            val intent = Intent(applicationContext, PlayerActivity::class.java)

            intent.putExtra("streamUrl", streamEditText.text.toString())
            intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

            applicationContext.startActivity(intent)
        }
    }
}
